proposals/0345-android-12-issues.md

```diff
@@ -0,0 +1,322 @@
+
+#### Developer Changes
+##### AndroidManifest.xml
+~~~ xml
+<activity
+    android:name="com.smartdevicelink.transport.USBAccessoryAttachmentActivity"
+    android:exported="true" <!--New Addition-->
+    android:launchMode="singleTop">
+    <intent-filter>
+        <action android:name="android.hardware.usb.action.USB_ACCESSORY_ATTACHED" />
+    </intent-filter>
+    <!-- ... -->
+</activity>
+
+~~~
+
+### PendingIntent Mutable Flag
+
+In Android 12, you must specify the mutability of each PendingIntent object that your app creates. This will impact the notifications that the `SdlRouterService` is trying to display, As we do not need to update these intents at any point we can flag them with `FLAG_IMMUTABLE`.
+
+#### Library Changes
+##### SdlRouterService.java
+~~~ java
+private void enterForeground(String content, long chronometerLength, boolean ongoing) {
+    //...
+    int flag = android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.S ? PendingIntent.FLAG_IMMUTABLE : 0;
+    PendingIntent pendingIntent = PendingIntent.getActivity(this, 0, intent, flag);
+    //...
+}
+~~~
+
+### Service Notification Delays
+
+Starting in Android 12 when a service tries to present a notification, Android may delay showing the notification for up to 10 seconds. This is to try to allow the service to complete before the notification is presented. If we have connected to a system before we may want to present the `SdlRouterService` notifications immediately. App developers may also want to display the notifications related to the `SdlService` immediately. This can be achieved by setting the foregroundServiceBehavior flag to `Notification.FOREGROUND_SERVICE_IMMEDIATE`.
+
+In the event this is the first time the `SdlRouterService` is trying to connect to a system or if the `SdlRouterService` has seen but not connected with the system we will not use this flag and allow Android to hide the notification.
+
+#### Library Changes
+##### SdlRouterService.java
+~~~ java
+private void enterForeground(String content, long chronometerLength, boolean ongoing) {
+    //...
+    Notification.Builder builder;
+    //...
+    if (hasConnectedBefore && android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+        builder.setForegroundServiceBehavior(Notification.FOREGROUND_SERVICE_IMMEDIATE);
+    }
+    //...
+}
+~~~
+
+## Potential downsides
+
+### Foreground Services
+
+Using `PendingIntents` to start the apps `SdlService` from the `SdlRouterService` creates a potential security risk as the apps `SdlService` will now be required to have the `android:exported` attribute set to true. This could expose apps to have their `SdlService` be started by apps that are not SDL certified. 
```

Comment:
```suggestion
Using `PendingIntents` to start the app's `SdlService` from the `SdlRouterService` creates a potential security risk as the app's `SdlService` will now be required to have the `android:exported` attribute set to `true`. This could expose apps to have their `SdlService` be started by apps that are not SDLC certified. 
```